lib/_http_client.js

```diff
@@ -562,7 +562,13 @@ function tickOnSocket(req, socket) {
   socket.on('close', socketCloseListener);
 
   if (req.timeout) {
-    socket.once('timeout', () => req.emit('timeout'));
+    const emitRequestTimeout = () => req.emit('timeout');
+    socket.once('timeout', emitRequestTimeout);
+    req.on('response', (r) => {
```

Comment:
Can you call the argument `res` for consistency/parity?  Is there a reason to use .on instead of .once?  The timeout listener is installed with .once().